src/background/utils/db.js

```diff
@@ -405,7 +405,7 @@ async function readEnvironmentData(env) {
   if (badScripts.size) {
     reportBadScripts(badScripts);
   }
-  env[PROMISE] = null; // indicating it's been processed
+  // env[PROMISE] = null; // indicating it's been processed
```

Comment:
@tophf This may not be a good fix, need your advice here.

Reply:
Yeah, now after sleeping on it I understand how this may happen. Gonna edit this PR.

Reply:
Done.